online-workflow/src/main/java/eu/itesla_project/online/tools/PrintOnlineWorkflowMetricsTool.java

```diff
@@ -15,114 +19,150 @@
+                    .hasArg()
+                    .required()
+                    .argName("STEP")
+                    .build());
+            options.addOption(Option.builder().longOpt("state")
+                    .desc("the state id (if not specified all the metrics of all the states are printed, in CSV format)")
+                    .hasArg()
+                    .argName("STATE")
+                    .build());
+            options.addOption(Option.builder().longOpt("output-file")
+                    .desc("export to a file")
+                    .hasArg()
+                    .argName("FILE")
+                    .build());
+            options.addOption(Option.builder().longOpt("output-format")
+                    .desc("output formats available: " + PrintOnlineWorkflowUtils.availableTableFormatterFormats() + " (default is ascii)")
+                    .hasArg()
+                    .argName("FORMAT")
+                    .build());
+            return options;
+        }
+
+        @Override
+        public String getUsageFooter() {
+            return null;
+        }
+
+    };
+
+    @Override
+    public Command getCommand() {
+        return COMMAND;
+    }
+
+    @Override
+    public void run(CommandLine line) throws Exception {
+        OnlineConfig config = OnlineConfig.load();
+        try (OnlineDb onlinedb = config.getOnlineDbFactoryClass().newInstance().create()) {
+            String workflowId = line.getOptionValue("workflow");
+            OnlineStep step = OnlineStep.valueOf(line.getOptionValue("step"));
+
+            Path outputFile = (line.hasOption("output-file")) ? Paths.get(line.getOptionValue("output-file")) : null;
+            String outputFormat = (line.hasOption("output-format")) ? line.getOptionValue("output-format") : "ascii";
+
+            TableFormatterConfig tableFormatterConfig = TableFormatterConfig.load();
+
+            if (line.hasOption("state")) {
+                Integer stateId = Integer.parseInt(line.getOptionValue("state"));
+                Map<String, String> metrics = onlinedb.getMetrics(workflowId, stateId, step);
+                if ((metrics == null) || (metrics.keySet().isEmpty())) {
+                    System.err.println("\nNo metrics for workflow " + workflowId + ", step " + step.name() + " and state " + stateId);
+                } else {
+                    String[] headers = new String[metrics.keySet().size() + 1];
+                    headers[0] = "state";
+                    int i = 1;
+                    for (String parameter : metrics.keySet()) {
+                        headers[i] = parameter;
+                        i++;
+                    }
+                    LinkedList<Column> columns = Arrays.stream(headers)
+                            .map(Column::new)
+                            .collect(Collectors.toCollection(LinkedList::new));
```

Comment:
what I wanted to convey to the 'reader' is that the list of columns is a sorted list, built in the same order as the attributes in the header from the stream